operator/internal/manifests/gateway_test.go

```diff
@@ -279,3 +279,44 @@ func TestBuildGateway_WithExtraObjectsForTenantMode_ReplacesIngressWithRoute(t *
 	require.NotContains(t, kinds, "*v1.Ingress")
 	require.Contains(t, kinds, "*v1.Route")
 }
+
+func TestBuildGateway_WithTLSProfile(t *testing.T) {
+	objs, err := BuildGateway(Options{
+		Name:      "abcd",
+		Namespace: "efgh",
+		Gates: configv1.FeatureGates{
+			LokiStackGateway: true,
+			HTTPEncryption:   true,
+			TLSProfile:       string(configv1.TLSProfileOldType),
+		},
+		TLSProfileSpec: configv1.TLSProfileSpec{
+			MinTLSVersion: "min-version",
+			Ciphers:       []string{"cipher1", "cipher2"},
+		},
+		Stack: lokiv1.LokiStackSpec{
+			Template: &lokiv1.LokiTemplateSpec{
+				Gateway: &lokiv1.LokiComponentSpec{
+					Replicas: rand.Int31(),
+				},
+			},
+			Tenants: &lokiv1.TenantsSpec{
+				Mode: lokiv1.OpenshiftLogging,
+			},
```

Comment:
IMHO using per default `Tenants.Mode: OpenShiftLogging` is misleading for our code base. Because it emits both required containers it does not make it a good test case. I suggest to make a test table with tenant modes and drop the following assert for this test as we have an other test case namely `TestBuildGateway_HasConfigForTenantMode`:
```go
require.Len(t, d.Spec.Template.Spec.Containers, 2)
```